File: code/python/isValid.py
class Solution:
    def isValid(self, s: str) -> bool:
        lst = []
        for i in range(len(s)):
            if s[i] == '(':
                lst.append(')')
            elif s[i] == '{':
                lst.append('}')
            elif s[i] == '[':
                lst.append(']')
            elif (len(lst) == 0 or lst.pop() != s[i]):
                return False
        return len(lst) == 0

File: code/java/isValid.java
class Solution {
    public boolean isValid(String s) {
        Stack stack = new Stack();
        char c;
        for(int i=0;i<s.length();i++){
            c = s.charAt(i);
            if(c == '{')
                stack.push('}');
            else if(c == '[')
                stack.push(']');
            else if(c ==  '(')
                stack.push(')');
            else if(stack.isEmpty() || (Character)stack.pop() != c)
                return false;
            
        }  
        return stack.isEmpty();
    }
}
